File: app/src/main/java/com/prey/actions/battery/Battery.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.battery;

public class Battery {

    private int health;
    private int iconSmall;
    private int level;
    private int plugged;
    private boolean present;
    private int scale;
    private int status;
    private String technology;
    private int temperature;
    private int voltage;
    private boolean charging;

    public boolean isCharging() {
        return charging;
    }

    public void setCharging(boolean charging) {
        this.charging = charging;
    }

    public int getHealth() {
        return health;
    }

    public void setHealth(int health) {
        this.health = health;
    }

    public int getIconSmall() {
        return iconSmall;
    }

    public void setIconSmall(int iconSmall) {
        this.iconSmall = iconSmall;
    }

    public int getLevel() {
        return level;
    }

    public void setLevel(int level) {
        this.level = level;
    }

    public int getPlugged() {
        return plugged;
    }

    public void setPlugged(int plugged) {
        this.plugged = plugged;
    }

    public boolean isPresent() {
        return present;
    }

    public void setPresent(boolean present) {
        this.present = present;
    }

    public int getScale() {
        return scale;
    }

    public void setScale(int scale) {
        this.scale = scale;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public String getTechnology() {
        return technology;
    }

    public void setTechnology(String technology) {
        this.technology = technology;
    }

    public int getTemperature() {
        return temperature;
    }

    public void setTemperature(int temperature) {
        this.temperature = temperature;
    }

    public int getVoltage() {
        return voltage;
    }

    public void setVoltage(int voltage) {
        this.voltage = voltage;
    }

}

File: app/src/main/java/com/prey/actions/battery/kotlin/Battery.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.actions.battery.kotlin

class Battery {
    private var health: Int = 0
    private var iconSmall: Int = 0
    private var level: Int = 0
    private var plugged: Int = 0
    private var isPresent: Boolean = false
    private var scale: Int = 0
    private var status: Int = 0
    private var technology: String? = null
    private var temperature: Int = 0
    private var voltage: Int = 0
    private var isCharging: Boolean = false

    fun isCharging(): Boolean {
        return isCharging
    }

    fun setCharging(isCharging: Boolean) {
        this.isCharging = isCharging
    }

    fun getHealth(): Int {
        return health
    }

    fun setHealth(health: Int) {
        this.health = health
    }

    fun getIconSmall(): Int {
        return iconSmall
    }

    fun setIconSmall(iconSmall: Int) {
        this.iconSmall = iconSmall
    }

    fun getLevel(): Int {
        return level
    }

    fun setLevel(level: Int) {
        this.level = level
    }

    fun getPlugged(): Int {
        return plugged
    }

    fun setPlugged(plugged: Int) {
        this.plugged = plugged
    }

    fun isPresent(): Boolean {
        return isPresent
    }

    fun setPresent(isPresent: Boolean) {
        this.isPresent = isPresent
    }

    fun getScale(): Int {
        return scale
    }

    fun setScale(scale: Int) {
        this.scale = scale
    }

    fun getStatus(): Int {
        return status
    }

    fun setStatus(status: Int) {
        this.status = status
    }

    fun getTechnology(): String {
        return technology!!
    }

    fun setTechnology(technology: String?) {
        this.technology = technology
    }

    fun getTemperature(): Int {
        return temperature
    }

    fun setTemperature(temperature: Int) {
        this.temperature = temperature
    }

    fun getVoltage(): Int {
        return voltage
    }

    fun setVoltage(voltage: Int) {
        this.voltage = voltage
    }
}